src/System.Private.CoreLib/shared/System/Char.cs

```diff
@@ -75,20 +75,20 @@ namespace System
         // Return true for all characters below or equal U+00ff, which is ASCII + Latin-1 Supplement.
         private static bool IsLatin1(char ch)
         {
-            return (ch <= '\x00ff');
+            return (uint)ch <= '\x00ff';
```

Comment:
You may be able to shave a bounds check by comparing with 0x100 here (have not verified it).

Reply:
Tried it, didn't help. Also tried to switch the order. I think there was some special way to write this comparison so that Roslyn encodes it in a way that jit recognises (it's used in number parsing if remember correctly).

Reply:
@pentp How do those casts to `uint` help?  Disassembly seems almost identical: [SharpLab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgBgARYCMA3ALABQWAzAQEx4DCFA3hXuwYQGx7AD2fADZ4AggAoAxgAsICPNICUbDq3Id1BAOx5JUvAB4AvHgDkAHQAeOHADMbJhWTUcAvnmXsiPfkLwAhXVl5KSVndlUNDixtMQBXAEsAOxgFaQNjcytbeyd1Fw9ObwFhEXEifChQ9QjIrR0oAG0cAF1000trOwdc13cwwt5i/wDyvEqCmsjonQTkhUaWtszOnIL88nWgA=).

Reply:
I was hoping they would help avoid a range check, but it didn't work.